### PR TITLE
Added option to include page body in crawl results

### LINKED ISSUE #8: Feature: Add option to return the crawled website body in the response
Currently we do not return the html body from the crawled sites. We only returns the links we find.

- Should have a flag to toggle this option
- Default set to False
- Modify the Json response to have keys `['urls', 'body']`

Eg:

```
{
    "http://github.com": {
        "urls": [
            "http://github.com/",
            "https://githubuniverse.com/",
            "..."
        ],
    "https://github.com/solutions/ci-cd": {
        "urls": [
            "https://github.com/solutions/ci-cd/",
            "https://githubuniverse.com/",
            "..."
        ]
      }
    }
}
```

This is a feature to return the html body as well. And the result should look look like this.

```
{
    "http://github.com": {
        "urls": [
            "http://github.com/",
            "https://githubuniverse.com/",
            "..."
        ]
        "body": "<html>stuff</html>",
    "https://github.com/solutions/ci-cd": {
        "urls": [
            "https://github.com/solutions/ci-cd/",
            "https://githubuniverse.com/",
            "..."
        ],
         "body": "<html>other stuff</html>",
      }
    }
}
```

### PR DESCRIPTION
**Pull request by Mews**
Closes #8 

**Changes**
- Added `include_body` argument in the `Spider` class. This is a boolean that defaults to false. When set to true, the body of the crawled pages will be included in `crawl_result`.
- In `Spider.crawl` added the code to add the body to `crawl_result` from `soup.html`.
- Added a test for this feature.

Right now the body of the page is added regardless of wether it finds links inside it or not! This just felt like the most expectable behavior, but let me know if I should change it. 
Also there are no verbose prints, I didn't find it necessary but let me know if I should add some 👍

### CODE REVIEW

**Comment by Mews** [comment]
Oh right I guess this new "body" field doesn't match the type hint for `crawl_result` :P
Should it be `Dict[str, Dict[str, Union[List[str], str]]]`?

File: tiny_web_crawler/crawler.py [hunk 1]
```diff
@@ -146,6 +148,9 @@ def crawl(self, url: str) -> None:
         links = soup.body.find_all('a', href=True) if soup.body else []
         self.crawl_result[url] = {'urls': []}
 
+        if self.include_body:
+            self.crawl_result[url]['body'] = str(soup.html)
```
Comment (indrajithi) [inline tiny_web_crawler/crawler.py:152]: we can use `str(soup)` here?

Comment (Mews) [reply tiny_web_crawler/crawler.py:152]: Yeah it does do the same, I just did `soup.html` to be sure, but I can change it if you preffer.

**Comment by Mews** [comment]
@indrajithi I'm not quite sure how to do the type hints for the `crawl_result` variable now that it has this new `body` field :/

File: tiny_web_crawler/crawler.py (hunk 1 above)
Comment (indrajithi) [reply tiny_web_crawler/crawler.py:152]: `str(soup.html)` will give us the content of the `<html>` tag and its children. `str(soup)` will give us the complete HTML document as a string including any `<!DOCTYPE>` declaration.

Comment (Mews) [reply tiny_web_crawler/crawler.py:152]: Ah true, ok changed it

**Comment by indrajithi** [comment]
> @indrajithi I'm not quite sure how to do the type hints for the `crawl_result` variable now that it has this new `body` field :/

`self.crawl_result: Dict[str, Dict[str, Union[List[str], str]]] = {}` Does this not work?

**Comment by indrajithi** [comment]
If the type hint for crawl_result is not working, we can just set it to a basic dict or override/suppress checking that case and move on.

**Comment by Mews** [comment]
Alright I'm on my phone right now but I'll get to it when I get home :+1:

**Comment by Mews** [comment]
> > @indrajithi I'm not quite sure how to do the type hints for the `crawl_result` variable now that it has this new `body` field :/
> 
> `self.crawl_result: Dict[str, Dict[str, Union[List[str], str]]] = {}` Does this not work?

Nope that's what raised the error on the ci.
I'll open an issue about it so that it can be dealt with later.

**Comment by Mews** [comment]
@indrajithi Ok I just introduced a temporary fix, I set the type hint to `Dict[str, Any]`, so you can rerun the ci and merge if everything passes. I'll open the issue now.

**Pull request merged**